src/EditorFeatures/Next/Structure/BlockContextProvider.cs

```diff
@@ -135,16 +136,24 @@ private ITextBuffer CreateProjectionBufferForBlockHeaders()
 
                 for (var i = 0; i < blockTags.Count; i++)
                 {
+                    var blockTag = blockTags[i];
+                    var fullStatementSpan = blockTag.StatementSpan;
+                    
+                    if (blockTag.Parent != null &&
+                        textSnapshot.AreOnSameLine(fullStatementSpan.Start, blockTag.Parent.StatementSpan.Start))
```

Comment:
It could be.  But this code only runs on hover.  And there should always be <10 items for 99.9% of all cases.  So hyperoptimizing just doesn't seem worth it.